src/wire/ipv4.rs

```diff
@@ -120,6 +120,45 @@ impl Cidr {
         self.prefix_len
     }
 
+    /// Return the netmask address of this IPv4 CIDR block.
+    pub fn netmask(&self) -> Address {
+        let mut addr = [0u8; 4];
+        if self.prefix_len >= 8 {
+            for idx in 0usize..((self.prefix_len - self.prefix_len % 8) / 8) as usize {
```

Comment:
Your computations are incorrect for prefix lengths that are not a multiple of 8. There is no requirement that prefix length be a multiple of 8.

Reply:
just do XOR ?

Reply:
I don't see how this helps. I've commented below on a test to show how this method should work.

Reply:
now should be correct。